vnext/Desktop.UnitTests/JsiRuntimeUnitTests.cpp

```diff
@@ -0,0 +1,1011 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
The file this is pulled from was a source copy of a JSI unit test harness from FB. https://github.com/facebook/react-native/blob/master/ReactCommon/jsi/jsi/test/testlib.cpp

They're already packaged in ReactCommon and the runtime should is pluggable. We can get some more coverage along with additional future coverage if we use that. It seems like we could also supplement the existing stuff and extend it instead of forking it.

Reply:
Note, those tests were already forked, as they were manually copied into our code base.
The reason this PR ported them to MSUT was the CI loops not actually running the JSI tests. Porting them allowed to drop the JSI test project and actually get the coverage.

That said, creating a "reference" MSBuild project (see ReactCommon.vcxproj) for this purpose could also work.
I'll try that approach to remove all forking of these tests.

Reply:
Thanks for taking the time to do this. I know this was already copied and slightly forked, but it will be nice to follow the "boyscout rule" here and leave things better than we found them. Technical issues tend to not get fixed if we leave existing issues when making changes.

Reply:
Resolved. I have turned the single file fork project JSI.Desktop.UnitTests into a test project aiming to run all GTest tests from ReactCommon.